src/node_contextify.cc

```diff
@@ -951,6 +952,125 @@ class ContextifyScript : public BaseObject {
 };
 
 
+void ContextifyContext::CompileFunction(
+    const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+  Local<Context> context = env->context();
+
+  // Argument 1: source code
+  CHECK(args[0]->IsString());
+  Local<String> code = args[0].As<String>();
+
+  // Argument 2: filename
+  CHECK(args[1]->IsString());
+  Local<String> filename = args[1].As<String>();
+
+  // Argument 3: line offset
+  CHECK(args[2]->IsNumber());
+  Local<Integer> line_offset = args[2].As<Integer>();
+
+  // Argument 4: column offset
+  CHECK(args[3]->IsNumber());
+  Local<Integer> column_offset = args[3].As<Integer>();
+
+  // Argument 5: cached data (optional)
+  Local<Uint8Array> cached_data_buf;
+  if (!args[4]->IsUndefined()) {
+    CHECK(args[4]->IsUint8Array());
```

Comment:
This should accept all ArrayBuffer view types, and ArrayBuffer itself, consistent with other newer APIs. Hint: use `Buffer::HasInstance()`

Reply:
I am kinda split if I should use Buffers or stick with what we have, also because it's consistent with `Script::New`. @hashseed @addaleax would love to hear your thoughts on this.

Reply:
I’d stick with consistency with `Script::New`, and maybe relax this in a later PR for both of them?

Reply:
Sounds great to me. @TimothyGu wdyt?

Reply:
Sure.